tests/FilePoolT.cpp

```diff
@@ -0,0 +1,327 @@
+// SPDX-License-Identifier: BSD-2-Clause
+
+// This code is part of the sfizz library and is licensed under a BSD 2-clause
+// license. You should have receive a LICENSE.md file along with the code.
+// If not, contact the sfizz maintainers at https://github.com/sfztools/sfizz
+
+#include "AudioSpan.h"
+#include "absl/types/span.h"
+#include "sfizz/Synth.h"
+#include "sfizz/FilePool.h"
+#include "TestHelpers.h"
+#include "catch2/catch.hpp"
+#include <algorithm>
+#include <vector>
+#include <chrono>
+#include <thread>
+#include <memory>
+#include <functional>
+#include <atomic>
+
+using namespace Catch::literals;
+
+static void wait_ms(int ms)
+{
+    std::this_thread::sleep_for(std::chrono::milliseconds(ms));
+}
+
+TEST_CASE("[FilePool] Ready Mutex")
+{
+    struct ProcessData
+    {
+        bool dummy;
+        sfz::FileData fileData {};
+        sfz::FilePool filePool {dummy};
+        void prepare()
+        {
+            wait_ms(200);
+            fileData.initWith(sfz::FileData::Status::Preloaded, sfz::FileData());
+        }
+        void wait()
+        {
+            fileData.addSecondaryOwner(&filePool);
+        }
+    };
+    ProcessData p;
+    std::thread t1([&p] { p.wait(); });
+    std::thread t2([&p] { p.prepare(); });
+    t1.join();
+    t2.join();
```

Comment:
Where these the reasons for hanging tests? We could also time out on a condition variable shared by the threads. It's a bit more verbose though.

Reply:
I added this for my local tests. I agree for that time out should be implemented.